Storage/SQL/SQLiteBookmarksSyncing.swift

```diff
@@ -490,6 +501,22 @@ open class SQLiteBookmarkBufferStorage: BookmarkBufferStorage {
                 return false
             }
 
+            // Delete pending deletions of items we just received.
```

Comment:
Also worth noting that there are two approaches here:

1. Keep local deletions for non-deleted remote changes. You might rename a bookmark on your desktop after you deleted it on your phone, and when your phone comes online it'll remove the bookmark from your desktop.
2. Discard local deletions for both remote deletions and remote changes. Your phone won't try to win any races at all.

Here you've implemented the second, which is simpler. It's a good idea to document that choice.